src/multiconfigwidget.h

```diff
@@ -11,32 +10,44 @@
 #include <KConfigGroup>
 #include <QWidget>
 
-class QComboBox;
-class QPushButton;
-
+namespace Ui {
+class MultiConfigWidget;
+}
+
+/** this widget allows fast switching between different configurations
+ * of m_child
+ * saveConfig will automatically save all changes to m_group
+ * use setChildWidget for an unparented one and moveWidgetToChild for a parented one
+ *
+ * ATTENTION: don't connect textChanged, etc to saveCurrentConfig, use returnPressed, ... otherwise saveCurrentConfig
```

Comment:
can't you catch this with a simple bool flag, which you set to true when saving/restoring, and if so, you don't trigger another restore while that is set to true?